### PR TITLE
add the plugin the 3.6 way

### PR DESCRIPTION
**Pull request by highstrike**
merging this pull request will increase cakephp's requirement to 3.6+

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+use Cake\Core\BasePlugin;
+use Cake\Routing\RouteBuilder;
+use Cake\Routing\Route\DashedRoute;
+
+/**
+ * Backend Plugin
+ *
+ * @author Flavius
+ * @version 1.0
```
Comment (dereuromark) [inline src/Plugin.php]: Should we keep these tags here?

File: src/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+class Plugin extends BasePlugin
+{
+    /**
+     * Plugin name
```
Comment (dereuromark) [inline src/Plugin.php]: We can remove this name.

File: src/Plugin.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public function routes($routes)
+    {
+        $routes->plugin(
+            'TestHelper',
+            ['path' => '/test-helper'],
+            function (RouteBuilder $routes) {
+                $routes->connect('/', ['controller' => 'TestHelper', 'action' => 'index']);
+
+                $routes->fallbacks(DashedRoute::class);
```
Comment (dereuromark) [inline src/Plugin.php]: Should we keep this empty, let the project decide the default?

File: README.md [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 Browser based Addons for your test driven development.
 
-Note: This branch is for CakePHP 3.5+.
+Note: This branch is for CakePHP 3.6+.
```
Comment (dereuromark) [inline README.md]: Why? If we do that we would need to increase composer, etc, too.

File: src/Plugin.php (hunk 1 above)
Comment (highstrike) [reply src/Plugin.php]: I don't care about the tags, it was copy/pasta from my backend plugin

File: src/Plugin.php (hunk 2 above)
Comment (highstrike) [reply src/Plugin.php]: ok

File: src/Plugin.php (hunk 3 above)
Comment (highstrike) [reply src/Plugin.php]: ok

File: README.md (hunk 4 above)
Comment (highstrike) [reply README.md]: cake 3.5 doesn't have the `BasePlugin.php` file in the core, this feature is only available since 3.6

Comment (dereuromark) [reply README.md]: But if we keep also old style both works?
I am ok with 3.6 only though. Just needs composer constraint.

Comment (highstrike) [reply README.md]: I would suggest increasing the constraints to require php 7.1 at least and cakephp to 3.6, yeah

Comment (highstrike) [reply README.md]: ok, well like i said, if you merge this you increase the requirement up to 3.6+, so you decide :)

**Comment by dereuromark** [comment]
Added to master, thx.

**Pull request closed without merging**